service/debugger/debugger.go

```diff
@@ -1467,6 +1467,65 @@ func (d *Debugger) Functions(filter string) ([]string, error) {
 	return funcs, nil
 }
 
+func traverse(t proc.ValidTargets, f *proc.Function, depth int, FollowCalls int) ([]string, error) {
+	if depth > FollowCalls {
+		return nil, nil
+	}
+	funcs := []string{}
+	funcs = append(funcs, f.Name)
+	text, err := proc.Disassemble(t.Memory(), nil, t.Breakpoints(), t.BinInfo(), f.Entry, f.End)
+	if err != nil {
+		return nil, fmt.Errorf("disassemble failed with error %s", err.Error())
+	}
+	depth = depth + 1
+	for _, instr := range text {
+		if instr.IsCall() && instr.DestLoc != nil && instr.DestLoc.Fn != nil {
+			cf := instr.DestLoc.Fn
+			if cf.Name == f.Name {
+				continue
+			}
+			if depth <= FollowCalls {
+				children, err := traverse(t, cf, depth, FollowCalls)
+				funcs = append(funcs, children...)
+				if err != nil {
+					return nil, fmt.Errorf("traverse failed with error %s", err.Error())
+				}
+			}
+		}
+	}
+	return funcs, nil
+
+}
+
+// FunctionsDeep returns a list of functions and it's children upto required depth indicated by FollowCalls in the target process.
+func (d *Debugger) FunctionsDeep(filter string, FollowCalls int) ([]string, error) {
```

Comment:
This should probably be a change to Debugger.Functions rather than a second copy with extra stuff.